docs/sources/developers/plugins/migration-guide.md

```diff
@@ -4,77 +4,418 @@ title = "Plugin migration guide"
+
+Update your `plugin` package to use the new plugin sdk.
+
+```go
+// before
+package plugin
 
-It's written for:
+import (
+  "context"
 
-- Plugin authors who want to migrate their plugins to Grafana 7.0+.
-- Plugin users who are using custom plugins and want to know whether they can upgrade to Grafana 7.0 without losing functionality.
+  "github.com/grafana/grafana_plugin_model/go/datasource"
+  "github.com/hashicorp/go-hclog"
+)
+
+func NewSampleDatasource(pluginLogger hclog.Logger) (*SampleDatasource, error) {
+	return &SampleDatasource{
+		logger: pluginLogger,
+	}, nil
+}
+
+type SampleDatasource struct{
+  logger hclog.Logger
+}
 
-> If you've successfully migrated your plugin from Angular to React, please [submit an issue on GitHub](https://github.com/grafana/grafana/issues/new?title=Docs%20feedback:%20/developers/plugins/migration-guide.md) and share your experiences with us so that we can improve this guide!
+func (d *SampleDatasource) Query(ctx context.Context, tsdbReq *datasource.DatasourceRequest) (*datasource.DatasourceResponse, error) {
+  d.logger.Info("QueryData called", "request", req)
+  // logic for querying your datasource.
+}
+
+// after
+package plugin
+
+import (
+  "context"
+
+  "github.com/grafana/grafana-plugin-sdk-go/backend"
+  "github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+  "github.com/grafana/grafana-plugin-sdk-go/backend/log"
+  "github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+func NewSampleDatasource(_ backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+	return &SampleDatasource{}, nil
+}
+
+type SampleDatasource struct{}
+
+
+func (d *SampleDatasource) Dispose() {
+	// Clean up datasource instance resources.
+}
+
+func (d *SampleDatasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+  log.DefaultLogger.Info("QueryData called", "request", req)
+  // logic for querying your datasource.
+}
+
+func (d *SampleDatasource) CheckHealth(_ context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+  log.DefaultLogger.Info("CheckHealth called", "request", req)
+  // The main use case for these health checks is the test button on the
+  // datasource configuration page which allows users to verify that
+  // a datasource is working as expected.
+}
+```
 
-## What's new in Grafana 7.0?
+### Unsigned backend plugins will not be loaded
+
+Before you can run a backend plugin on Grafana installations, you must sign the plugin. Grafana Labs strongly recommends you upload signed plugins since we cannot guarantee the integrity of unsigned plugins.
+
+For instructions on plugin signing, refer to the [Plugin signatures](https://grafana.com/docs/grafana/latest/developers/plugins/sign-a-plugin/#sign-a-plugin) topic.
+
+You can still run and develop an unsigned plugin by running your Grafana instance in [development mode](https://grafana.com/docs/grafana/latest/administration/configuration/#app_mode).
```

Comment:
I don't think we want to encourage anyone to run their Grafana instance in anything other than `production`. I think using [this](https://grafana.com/docs/grafana/latest/administration/configuration/#allow_loading_unsigned_plugins) config is the most appropriate way to go for now

Reply:
```suggestion
You can still run and develop an unsigned plugin by running your Grafana instance in [development mode](https://grafana.com/docs/grafana/latest/administration/configuration/#app_mode). Alternatively, you can use [this configuration setting.]({{< relref "../administration/#allow_loading_unsigned_plugins" >}})
```

Reply:
Yea, I think that is a good addition, thanks @wbrowne!